main.tf

```diff
@@ -5,12 +5,18 @@
 locals {
   sg_name_null_and_use_sg_true = var.existing_security_group_name == null && var.use_existing_security_group
   sg_name_set_and_use_sg_false = var.existing_security_group_name != null && !var.use_existing_security_group
-  no_sg_name_and_no_vpc_id     = var.existing_security_group_name == null && var.vpc_id == null
+  sg_id_null_and_use_sg_true   = var.existing_security_group_id == null && var.use_existing_security_group_id
+  sg_id_set_and_use_sg_false   = var.existing_security_group_id != null && !var.use_existing_security_group_id
+  no_sg_name_and_no_vpc_id     = (var.use_existing_security_group == false && var.vpc_id == null) || (var.use_existing_security_group_id == false && var.vpc_id == null)
```

Comment:
I think this logic might fail.

it will still trigger (be true) if use_existing_security_group == true and vpc_id == null (which is valid right?), because of the OR condition for the use_id being false?

For instance, this local will be true if:
vpc_id = null
use_existing_group = true
use_existing_id = false

I think you want to replace the OR with an AND?